master/internal/core.go

```diff
@@ -66,12 +66,12 @@ type Master struct {
 
 // fileExists checks if a file exists and is not a directory before we
 // try using it to prevent further errors.
-func fileExists(filename string) bool {
+func fileExists(filename string) (bool, error) {
 	info, err := os.Stat(filename)
 	if os.IsNotExist(err) {
-		return false
+		return false, nil
 	}
-	return !info.IsDir()
+	return !info.IsDir(), err
```

Comment:
We still can't call `info.IsDir()` when there is an error. See https://play.golang.org/p/Ju_2qGJqaNc

Reply:
:facepalm:  used aswitchcase here